cmd/snap-confine/snap-confine.c

```diff
@@ -243,6 +246,9 @@ int main(int argc, char **argv)
 		 * profile as well. */
 		sc_apply_global_seccomp_profile();
 	}
+#ifdef HAVE_SELINUX
+	sc_selinux_set_snap_execcon();
+#endif
```

Comment:
can we have  a note about that this is done both for confined and classic snaps, in any case the context of the snap != context of snap-confine

Reply:
Moved that around to be closed to sc_maybe_aa_change_onexec(). I see that the AppArmor support code is in libsnap-confine-private (?). I'd be happy to propose a follow-up PR moving selinux code around to that location.